File: flutter/lib/domain/repository/deck_repository.dart
import '../entities/deck.dart';

abstract class DeckRepository {
  Future<int> addDeck(Deck deck);
  Future<void> updateDeck(Deck updatedDeck);

  Future<void> deleteDeckById(Deck deck);
  Future<void> deleteDeck(Deck deck);
  Future<void> deleteAllDecks();

  Future<List<Deck>> getAllDecks();
  Future<List<Deck>> getAllDecksWithFlashcards();
  Future<Deck> getDeckById(int deckId);
}

File: android/app/src/main/java/com/synergyboat/flutterai/domain/repository/DeckRepository.kt
package com.synergyboat.flutterai.domain.repository
import com.synergyboat.flutterai.domain.entities.Deck

interface DeckRepository {
    suspend fun getAllDecks(): List<Deck>
    suspend fun createDeck(deck: Deck): Deck
    suspend fun deleteDeck(id: Int)
    suspend fun updateDeck(deck: Deck): Deck
    suspend fun getDeckById(id: Int): Deck?
}